samples/net/echo_server/src/echo-server.c

```diff
@@ -144,3 +141,61 @@ void main(void)
 		stop_udp();
 	}
 }
+
+static void do_service(k_work_handler_t handler)
+{
+	static struct k_work work;
+
+	k_work_init(&work, handler);
+
+	k_work_submit(&work);
+}
+
+static void event_handler(struct net_mgmt_event_callback *cb,
+			  u32_t mgmt_event, struct net_if *iface)
+{
+	if (mgmt_event == NET_EVENT_L4_READY) {
+		NET_INFO("Network connectivity ready.");
+
+		do_service(start_udp_and_tcp);
+
+		return;
+	}
+
+	if (mgmt_event == NET_EVENT_L4_DISCONNECTED) {
+		NET_INFO("Network disconnected.");
+
+		do_service(stop_udp_and_tcp);
+
```

Comment:
Should the application also know which interface disconnected, so that echo_server can undo whatever internal variables it set up for a particular interface?